pex/venv/pex.py

```diff
@@ -91,6 +91,40 @@ def _copytree(
             return
 
 
+def _create_shebang(
+    venv_python, # type: str
+):
+    # type: (...) -> str
+    """Return the shebang to be used in the venv pex invoker script.
+
+    When the shebang is longer than the common linux length limit of 128 Bytes we try
+    to replace the shebang with a snippet of shell script which will invoke the correct
+    python interpreter from a path relative to the script itself, which avoids the need
+    to encode the long interpreter as a shebang.
+    """
+
+    # This multiline shebang locates the interpreter in the venv relative to the file this shebang is placed in.
+    # For example the script is usually ~/.pex/venvs/<sha>/<sha>/pex and the venv bin dir is ~/.pex/venvs/<sha>/<sha>/bin/.
+    # The exec command re-execs the venv pex script with the venv interpreter. The unfortunate
+    # trick here is that the shell exec command and its arguemetns are wrapped in quotes so the python
+    # interpreter treats it as a static string when executing the script, essentially ignoring it.
+    long_shebang_exec = '#!{sh_path}\n"exec" "`dirname $0`/bin/{interpreter_basename}" "-sE" "$0" "$@"'
+
+    shebang = "#!{} -sE".format(venv_python)
+    if len(shebang) > 128:
```

Comment:
And this is even less generally valid.

Reply:
Yeah, it's not always the limit. Could check for darwin platform here, but I figured might as well no throw in platform logic.

Reply:
My point is more about the fact you're checking on the wrong machine. This PEX build machine is generally not the deployment machine. IOW this may work for your case (say PEXes are built in CI on machines / images that match prod deployment machines / images), but it does not generally work and the feature is not behind a flag - you're subjecting every Pex user to it.